### PR TITLE
Added UI and backend for notetaking

### PR DESCRIPTION
**Pull request by Parth18Shah**
- Added UI page for notetaking
- Added an endpoint to save those notes
- Integrated TipTap editor to take the notes
- Added a Menu bar to the editor to perform various operations on the text
- Added comments (wherever necessary) to improve readability
- Used the debounce hook to add delays between every save, hence improving the website's performance

### CODE REVIEW

File: src/components/TipTapEditor.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+import axios from 'axios';
+import { LoaderPinwheel } from 'lucide-react';
+type Props = {
+    note: any;
+}
+/*
+    Description ->  React Component for creating a TipTap Editor.
+*/ 
+const TipTapEditor = ({ note }: Props) => {
+    const [content, setContent] = React.useState(note.content || 'Start typing...');
```
Comment (dsathe) [inline src/components/TipTapEditor.tsx:19]: Start Typing does not go away when I click to start typing. Can we make it go away when we click on text area ?
Just a suggestion nothing important.

File: src/lib/clerk-server.ts
Comment (dsathe) [inline src/lib/clerk-server.ts:1]: What is the use of this? Does any specific instance like debouncing require this?

**Comment by dsathe** [comment]
The code is working on my end. 
Able to install the packages and it worked as expected.
Lets sync up and commit the changes.

File: src/lib/clerk-server.ts
Comment (Parth18Shah) [reply src/lib/clerk-server.ts:1]: This is used to get the user details on the client side. Here, I am using it on the noteUI page (line 26) to get the user's name to display it on the note's page.

File: src/components/TipTapEditor.tsx (hunk 1 above)
Comment (Parth18Shah) [reply src/components/TipTapEditor.tsx:19]: It's not a placeholder but I could look into making it into one in the future.

**Review by dsathe — APPROVED** [review]
Changes looks good approving the PR.

**Pull request merged**